pkg/nsx/services/subnetport/builder.go

```diff
@@ -65,6 +65,24 @@ func (service *SubnetPortService) buildSubnetPort(obj interface{}, nsxSubnet *mo
 	}
 	namespace_uid := namespace.UID
 	tags := util.BuildBasicTags(getCluster(service), obj, namespace_uid)
+
+	var tagsFiltered []model.Tag
+	for _, tag := range tags {
+		if isVmSubnetPort && *tag.Scope == common.TagScopeNamespaceUID {
```

Comment:
Can we add some comments for the logic here, there are a lot of blocks for different conditions.

Reply:
fixed